libs/csight-core/src/bdsp/read.rs

```diff
@@ -56,10 +56,10 @@ fn read_party_pokemon(index: u8) -> Pk8 {
     read_pokemon_in_party(player_party, index)
 }
 
-fn read_wild_pokemon() -> Pk8 {
+fn read_encounter_pokemon(index: u8) -> Pk8 {
     // Wild Pokemon are set up in a fake party
     let battle_party = get_battle_setup_param().follow(0x58).follow(0x28);
-    read_pokemon_in_party(battle_party, 0)
+    read_pokemon_in_party(battle_party, index)
 }
```

Comment:
Let's expand upon this so the wild Pokemon count can be obtained:
```rust
fn read_pokemon_count_in_party(reader: DmntReader) -> u32 {
    let count = reader.read_offset(0x18);
    core::cmp::min(count, 6)
}

fn read_encounter_party() -> DmntReader {
    // Wild Pokemon are set up in a fake party
    get_battle_setup_param().follow(0x58).follow(0x28)
}

fn read_encounter_pokemon(index: u8) -> Pk8 {
    let encounter_party = read_encounter_party();
    read_pokemon_in_party(encounter_party, index)
}

fn read_encounter_pokemon_count() -> u32 {
    let encounter_party = read_encounter_party();
    read_pokemon_count_in_party(encounter_party)
}
```

This will be needed in the `c_api` module:
```rust
#[no_mangle]
pub unsafe extern "C" fn bdsp_read_encounter_pokemon_count() -> u32 {
    read_encounter_pokemon_count()
}
```

The c binding [to add under this line](https://github.com/zaksabeast/CaptureSight/blob/6e75131572769c6cfe687cbb6452b6bd079e64fd/libs/csight-core/c-export/bdsp.h#L39):
```c
u64 bdsp_read_encounter_pokemon_count();
```

And the cpp binding [to add under this line](https://github.com/zaksabeast/CaptureSight/blob/6e75131572769c6cfe687cbb6452b6bd079e64fd/libs/csight-core/c-export/bdsp.h#L64):
```cpp
u64 read_encounter_pokemon_count() { return bdsp_read_encounter_pokemon_count(); }
```